packages/block-editor/src/components/inserter/hooks/use-unsynced-patterns.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { parse } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import useBlockTypesState from '../hooks/use-block-types-state';
+
+export default function useUnsyncedPatterns(
+	rootClientId,
+	onInsert,
+	withBlocks
+) {
+	const [ unsyncedPatterns ] = useBlockTypesState(
```

Comment:
Isn't it a bit weird, to have `useBlockTypesState` responsible for returning the "unsynced" patterns? Should we move these to `usePatternsState` instead? which would probably ensure that we don't need a special case for unsynced pattern and the `useUnsyncedPatterns` hook is not needed anymore.

Reply:
This makes sense, but is a reasonable refactor, so have done it as a s[eparate PR here](https://github.com/WordPress/gutenberg/pull/51947).